src/backend/cdb/cdbpartition.c

```diff
@@ -5261,12 +5263,22 @@ get_part_rule(Relation rel,
 									pid2,
 									bExistError, bMustExist,
 									pSearch, pNode, sid1.data, &pNode2);
+			if (!prule2)
```

Comment:
I suggest such simple change
```suggestion
			if (!prule2)
				return NULL;
```